watch.js

```diff
@@ -453,32 +453,57 @@ function handleReply(self, msg) {
 	delete self.reply[msg.d.r];
 }
 
+function _convertObjToLevels(inputObj) {
+
+	if (!inputObj || (typeof inputObj !== 'object')) {
+		return inputObj;
+	}
+
+	var objWithLevels = {};
+
+	Object.keys(inputObj || {}).forEach(function(key) {
+
+		var path = key.split('/');
+
+		if (path.length === 1) {
+			objWithLevels[path[0]] = _convertObjToLevels(inputObj[key]);
+		} else {
+			var obj = {}, level = obj;
+			for (var i = 1; i < path.length-1; i++) {
+				level = level[path[i]] = {};
+			}
+			level[path[path.length-1]] = _convertObjToLevels(inputObj[key]);
+			objWithLevels[path[0]] = obj;
+		}
+	});
+
+	return objWithLevels;
+
+}
+
 function handleUpdate(self, path, newData) {
-	path = path.split('/');
 
+	var augmentedNewData = {};
+	augmentedNewData[path] = newData;
```

Comment:
Make a wrapper obj, which has the passed-in `path` as its only key, where the value is `newData`. Then toss it into the helper function `_convertObjToLevels`, which recursively makes levels for all keys (including nested ones) with path strings (i.e., strings with `"/"` in them).